activemq-artemis/templates/servicemonitor.yaml

```diff
@@ -0,0 +1,20 @@
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
```

Comment:
I'm super curious about this. I've cosidered in the past to use just an annotation and expect that prometheus should be configured outside. Do you find that to have a prometheus as a dependency of the chart is the best practice? I don't really know but haven seen something similar in most of the official charts I've checked.

Reply:
In our solution we use the latest Prometheus (Prometheus Operator) which requires that you use the ServiceMonitor instead of the annotations. The ServiceMonitor allows much finer-grained control over what gets scraped (including which ports, etc. otherwise the older annotation approach tried to scrape all ports for an annotated service). 

In terms of 'where does it live' - I agree with your caution. We're still investigating the various ways to package/bundle these types of artefacts with our solution (e.g. Grafana dashboards, Prometheus scrape configs, etc.). Right now, we have it as part of the Helm chart because it only makes sense to deploy the Artemis ServiceMonitor if you have Artemis installed. And Prometheus is already a standard part of our system, so we know it'll always be there.

We could definitely consider have it as a separate Helm chart, though. No strong feelings either way.